### ISSUE TITLE
--add-metadata / --embed-metadata from Python

### ISSUE THREAD

**Issue by suuuehgi** [yt-dlp/yt-dlp#2462]
### Checklist

- [X] I'm asking a question and **not** reporting a bug/feature request
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions including closed ones

### Question

Is there a way to use the `--embed-metadata` / `--add-metadata` flag from python?
If not, is there any plan on adding it?
It's not listet within [YoutubeDL.py](https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/YoutubeDL.py#L191)

Thank a lot!

**Comment by pukkandan** [comment]
https://github.com/yt-dlp/yt-dlp/issues/1901

**Comment by suuuehgi** [comment]
Thank's for the quick hint!

So there is no way to get to know the PP but to read their source code?

I'm currently using `FFmpegExtractAudioPP` and would just like to add metadata.

**Comment by pukkandan** [comment]
Yeah, the documentation of PPs are a bit lacking. It may be usefull to refer to this to see how CLI options translates to PPs. https://github.com/yt-dlp/yt-dlp/blob/88f23a18e0a0f5f18ed167b6a319c54255c5dd13/yt_dlp/__init__.py#L467-L613

You can also see each initializer for the arguments it takes: https://github.com/yt-dlp/yt-dlp/blob/88f23a18e0a0f5f18ed167b6a319c54255c5dd13/yt_dlp/postprocessor/ffmpeg.py#L670-L672

> For FFmpegMetadataPP, audio_only seems to be hard-coded to aac.

I don't understand...

**Comment by suuuehgi** [comment]
Thank you very much! I'm gonna have a look at it when I find some time.

> > For FFmpegMetadataPP, audio_only seems to be hard-coded to aac.
> 
> I don't understand...

I mean the following segment as m4a is limited to MPEG-4 Part 3, AAC and Apple Lossless but afaics in practice mainly used for AAC.

> https://github.com/yt-dlp/yt-dlp/blob/6970b6005e9c07c427d368bbe3f71f85878f325e/yt_dlp/postprocessor/ffmpeg.py#L679-L683

**Comment by pukkandan** [comment]
That's implementation detail
